src/viewer/core/ViewerWindowManager.C

```diff
@@ -4489,11 +4489,35 @@ ViewerWindowManager::UpdateColorTable(const std::string &ctName)
 //    Brad Whitlock, Wed Apr 30 09:58:46 PDT 2008
 //    Added tr().
 //
+//    Mark C. Miller, Fri Mar 17 15:14:30 PDT 2023
+//    Add APPLE-specific logic to address blank viewer windows (#18090)
 // ****************************************************************************
 
 void
 ViewerWindowManager::SetWindowLayout(const int windowLayout)
 {
+
+    // Stop gap to avert blank viewer windows (#18090)
+    // There is associated logic in rpc/ViewerMethods::DrawPlots which will
+    // trigger this logic twice upon **FIRST** draw. This causes window to
+    // repaint and not be blank. Attempts to use SetWindowSize() also fixed
+    // the blank window issue but due to internal inconsistency in what
+    // VisIt thinks the window size actually is failed to return the window
+    // to its original size.
+#if defined(__APPLE__)
+    static size_t count = 0;
+    if (windowLayout == -5 && count < 2)
+    {
+        static int origlo = layout;
```

Comment:
Ooohh...what a minute...if the original layout, `origlo` is also `2`, then this might result in a no-op. @cyrush do you have VisIt start up by default in layout `2`?

Reply:
I think `-noconfig` + new build works.

But not sure if that helps folks in the wild b/c they may have a config :-(

Reply:
Ok, but...what was your `config` layout set to? I mean if it just happened to match the hard-coded logic, that would explain why it didn't work...it could have been a no-op for you. Can you attach your `config` here for me to test?

Reply:
@cyrush I just remembered...to trigger this logic, you do actually need to plot something. It happens the *first time* anything is plotted.

Reply:
What entry does this correspond to in the `guiconfig` or `config` -- I don't see any `layout` entry in my config files.

Reply:
> What entry does this correspond to in the `guiconfig` or `config` -- I don't see any `layout` entry in my config files.

Hmm...than maybe you have default layout. That said, you do need to actually plot something to trigger the behavior. You can't just start VisIt and notice the window is in this odd-ball state.